SCMS.Portal.Tests.Unit/Services/Views/StudentViews/StudentViewServiceTests.Exceptions.Add.cs

```
@@ -0,0 +1,159 @@
+﻿// -----------------------------------------------------------------------
+// Copyright (c) Signature Chess Club & MumsWhoCode. All rights reserved.
+// -----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Moq;
+using SCMS.Portal.Web.Models.Foundations.Students;
+using SCMS.Portal.Web.Models.Views.StudentViews;
+using SCMS.Portal.Web.Models.Views.StudentViews.Exceptions;
+using Xunit;
+
+namespace SCMS.Portal.Tests.Unit.Services.Views.StudentViews
+{
+    public partial class StudentViewServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(StudentServiceValidationExceptions))]
+        public async Task ShouldThrowDependencyValidationExceptionOnAddIfStudentValidationErrorOccuredAndLogItAsync(
+            Exception studentServiceValidationException)
+        {
+            //given
+            StudentView someStudentView = CreateRandomStudentView();
+
+            var expectedDependencyValidationException =
+                new StudentViewDependencyValidationException(studentServiceValidationException);
+
+            this.studentServiceMock.Setup(service =>
+                service.AddStudentAsync(It.IsAny<Student>()))
+                    .ThrowsAsync(studentServiceValidationException);
```

We can throw the exception at the first dependency call. The scope of the test is to validate the exception handling and not concerned about where it is coming from.